recipe/2023-11-support-insecure-registries.md

```diff
@@ -0,0 +1,185 @@
+# Adding Support for publishing recipes to insecure registries
+
+* **Status**: Pending
+* **Author**: Vishwanath Hiremath (@vishwahiremat)
+
+## Overview
+
+Today we publish/pull recipes to/from the registry that is OCI compliant and only support secure connections i.e registries that are SSL/TLS enabled. And we don't have support to use insecure registry e.g locally hosted k3d registry. And we hit _"http: server gave HTTP response to HTTPS client"_ error if insecure registry is used. This document is centered around addressing this issue.
+
+## Terms and definitions
+
+| Term        | Definition |
+| :-- |:--|
+| insecure registry      | An insecure registry is the one not having either valid registry certificate or is not using TLS and does not use secure communication protocols like HTTPS. |
+| ORAS      | ORAS is the de facto tool for working with OCI Artifacts.https://oras.land/docs/ |
+| Recipes      | Recipes enable a separation of concerns between infrastructure operators and developers by automating infrastructure deployment. https://docs.radapp.dev/author-apps/recipes/ |
+
+## Objectives
+
+> **Issue Reference:** https://github.com/radius-project/radius/issues/6648
+
+### Goals
+
+-   Enable the support to publish and pull recipes using a non-secure registry.
```

Comment:
Can we explain why we are pursuing this goal? What blockage did we run into that made us think that we have to support insecure registries?

Reply:
Was it something in the pipeline?

Reply:
Or local testing?

Reply:
We ran into issue when we are working test framework for recipes repository where we wanted to publish recipes to local k3d managed registry and we were hitting this issue
![image](https://github.com/radius-project/design-notes/assets/100623239/d91d9b20-d5b9-4156-b4ef-89f9660322f2)